jupyter_book/sphinx.py

```diff
@@ -138,6 +140,21 @@ def debug_args():
             )
             app.config.latex_documents = new_latex_documents
 
+            # setting up sphinx-multitoc-numbering
+            if app.config["use_multitoc_numbering"]:
+                # if sphinx-external-toc is used
+                if "external_toc_path" in app.config:
+                    import yaml
+
+                    site_map = app.config.external_site_map
+                    site_map_str = yaml.dump(site_map.as_json())
+
+                    # only if there is atleast one numbered: true in the toc file
+                    if "numbered: true" in site_map_str:
+                        app.setup_extension("sphinx_multitoc_numbering")
+                else:
+                    app.setup_extension("sphinx_multitoc_numbering")
```

Comment:
@AakashGfude this will setup the extension even if `numbered: true` is not in the `site_map_str` given it is also in the `else` condition right?

Reply:
Good question. The else condition is for the case when `sphinx-external-toc` is not used. Should we set up the extension by default in those cases? Like for sphinx toctree directives.

Reply:
ah thanks @AakashGfude -- misread the indentation levels. 

In the `jupyter-book` context will that ever be the case? I think `sphinx-external-toc` will manage `_toc.yml` files always right?

Reply:
I think so too. but I saw this check here as well https://github.com/executablebooks/jupyter-book/blob/master/jupyter_book/config.py#L164 on the presence of `sphinx-external-toc`, so thought should handle that edge case when adding this extension.
But I think you are right as this specifically deals with toc. So `sphinx-external-toc` should be assumed to be present.

Reply:
Should I remove this `else` case?

Reply:
Let's leave it there -- sounds good @AakashGfude.